### ISSUE TITLE
[Bug]: When starting from VSCode extension, 'devtools: true' in launchOptions does nothing

### ISSUE THREAD

**Issue by karyon** [microsoft/playwright#29899]
### Version

1.42.1

### Steps to reproduce

1. clone https://github.com/karyon/playwright-examples
2. git checkout repro-devtools-doing-nothing
3. npm install, npx playwright install
4. Using the VSCode extension, run any test with "show browser" enabled

### Expected behavior

Browser opens and has devtools open

### Actual behavior

Browser opens but does not have devtools open

### Additional context

[browserType.launch() docs](https://playwright.dev/docs/api/class-browsertype#browser-type-launch)
[launchOptions docs](https://playwright.dev/docs/test-use-options#more-browser-and-context-options) say any browserType.launch() option ([docs](https://playwright.dev/docs/api/class-browsertype#browser-type-launch)) can be used
In https://github.com/microsoft/playwright/issues/28529#issuecomment-1845009195 someone is reporting this is working for them
In playwright.config.ts, uncomment `args: ['--auto-open-devtools-for-tabs']`, this does work.

### Environment

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
    Memory: 6.92 GB / 31.76 GB
  Binaries:
    Node: 18.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.21 - C:\Program Files\nodejs\yarn.CMD
    npm: 9.6.7 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.15.4 - C:\Program Files\nodejs\pnpm.CMD
  IDEs:
    VSCode: 1.85.2 - C:\Program Files\Microsoft VS Code\bin\code.CMD
  Languages:
    Bash: 5.2.26 - C:\Program Files\Git\usr\bin\bash.EXE
  npmPackages:
    @playwright/test: ^1.42.1 => 1.42.1
```

**Comment by mxschmitt** [comment]
Investigation: The `devtools: true` option does not get taken into account when using the VSCode extension.

When running the tests via the CLI, it works: `npx playwright test --project=chromium --headed api-mocking:5`.

**Comment by karyon** [comment]
Ah I forgot! When using  `chromium.launch` in test code, it works too.

**Comment by karyon** [comment]
I agree with @odinho in #30025. I used this option to debug code in page.evaluate. Putting breakpoints in VSCode in a page.evaluate doesn't seem to do anything. I'll have to continue using `args: ['--auto-open-devtools-for-tabs']` which seems like a poor workaround.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
